app/models/device_story/device.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module DeviceStory
+  class Device < ApplicationRecord
+    self.table_name = 'device_story_devices'
+
+    validates :device_urn, presence: true, uniqueness: true
+    validates :metadata, presence: true
+
+    before_save :update_from_metadata
+
+    private
+
+    def update_from_metadata
+      self.device_id = metadata['device']
+      self.last_location = "POINT(#{metadata['last_lon'] || 0} #{metadata['last_lat'] || 0})"
```

Comment:
I'm sure that we don't have a device at `0,0`.

However, `0,0` are valid coordinates. Is it possible to instead set `last_location` to null? In general, using valid values to indicate a lack of data causes confusion in systems.

Reply:
0,0 means that it is located in the middle of the ocean. I would consider it not valid. It indicated no GP coordinates are sent.

Reply:
A human being might look at `0,0` and understand that it's not valid. But a computer won't know that without special programming. There are many examples of problems caused by this, the most famous of which is this:

https://splinternews.com/how-an-internet-mapping-glitch-turned-a-random-kansas-f-1793856052

In computing, we already have standard concepts for the idea of "no data" -- `null`, `undefined`, `nil`, etc. I am not aware of any reason to make things more complicated by taking a real location and making it the "no data" location.

Reply:
The reason for not allowing 0,10 for me is that if a string of coordinates is being fed into a map in Grafana or Kibana, the software will show 00 and the cluster of the other coordinates. That means often as a wide scaling.

Reply:
I think we're agreeing with each other.

Reply:
I am not sure why only `0,0` is being mentioned. In the worst case, we could have `0,0`. But, how often?

We have `metadata` column. We should look at this column if we need to look seriously. I believe that `last_*` columns are for peeking without digging matadata.

Anyway, I changed to set `last_location` only if both `last_lat` and `last_lot` are present.